### PR TITLE
feat: Output record count metric from batch files insert

### PR DESCRIPTION
**Pull request by ReubenFrankel**
Bit of an opinionated change since it might be conflating batches/records, but it has been helpful for us to know how many rows (records) were created/updated from batch processing.

### CODE REVIEW

File: target_snowflake/sinks.py [hunk 1]
```diff
@@ -199,13 +199,16 @@ def insert_batch_files_via_internal_stage(
...
-                self.connector.copy_from_stage(
+                record_count = self.connector.copy_from_stage(
                     full_table_name=full_table_name,
                     schema=self.schema,
                     sync_id=sync_id,
                     file_format=file_format,
                 )
 
+            with self.record_counter_metric as counter:
+                counter.increment(record_count)
```
Comment (edgarrmondragon) [inline target_snowflake/sinks.py:210]: Wouldn't this add records to the count twice? Once when the record is processed in https://github.com/meltano/sdk/blob/409a40b48c442e1382611d3a69b2f95df2e073d3/singer_sdk/target_base.py#L362 and again when they're batched in https://github.com/MeltanoLabs/target-snowflake/blob/8919a663f8667136bd3b6a1e5bf1369fd46b26af/target_snowflake/sinks.py#L151-L154?

Comment (ReubenFrankel) [reply target_snowflake/sinks.py:210]: Hmm, I guess so. The intention of this change was to work with `BATCH` messages, but you're referring to when a user supplies `batch_config` to batch together `RECORD` data into files for insert via internal stage?

Comment (edgarrmondragon) [reply target_snowflake/sinks.py:210]: Maybe the full context of the function I linked is useful:

https://github.com/MeltanoLabs/target-snowflake/blob/7c9a1fb342ab098b43ad314427bd6bc9739690d8/target_snowflake/sinks.py#L109-L156

`RECORD` messages are processed by this logic, even with the default `batch_config`.

Comment (ReubenFrankel) [reply target_snowflake/sinks.py:210]: I see. So processing of `RECORD` and `BATCH` messages both end up calling `insert_batch_files_via_internal_stage` where I've made this change, except `RECORD` messages are already counted by the SDK before they are batched so would now get counted twice.

In spite of this issue, in principle how do you feel about this idea of emitting a record count when this target receives a `BATCH` message? Is it OK or does it conflate two concepts?

File: target_snowflake/sinks.py [hunk 2]
```diff
@@ -242,6 +244,9 @@ def process_batch_files(
                 msg,
             )
 
+        with self.record_counter_metric as counter:
+            counter.increment(record_count)
```
Comment (edgarrmondragon) [inline target_snowflake/sinks.py:248]: Yeah, this should only count records inserted from batch files!

**Review by edgarrmondragon — APPROVED** [review]

**Comment by edgarrmondragon** [comment]
Thanks @ReubenFrankel!

**Pull request merged**